compaction/compaction.cc

```diff
@@ -1766,7 +1770,7 @@ get_fully_expired_sstables(const table_state& table_s, const std::vector<sstable
     int64_t min_timestamp = std::numeric_limits<int64_t>::max();
 
     for (auto& sstable : overlapping) {
-        auto gc_before = sstable->get_gc_before_for_fully_expire(compaction_time);
+        auto gc_before = sstable->get_gc_before_for_fully_expire(compaction_time, cm.get_tombstone_gc_state());
         if (sstable->get_max_local_deletion_time() >= gc_before) {
```

Comment:
Why don't we get tombstone_gc_state using _table_s.get_compaction_strategy().get_tombstone_gc_state();?

I think we shouldn't pass compaction_manager around, it's like the highest layer in compaction layer and should never be passed down to any lower level entity.

Reply:
I'll give it a shot